compiler/Makefile

```diff
@@ -5,11 +5,13 @@ BAL_BUILD_FLAGS ?= --offline
 
 all: $(COMPILER_JAR)
 
-test:
+# need to run `bal build` after `bal test` since `bal test` removes the jar
```

Comment:
Another option is to trigger `bal build` from `nballerina/test/Makefile`. But since in future (hopefully) `bal test` will produce a jar. So we can mock that behavior for now.